### PR TITLE
feature server should send USER call to the sbc sip that is connect with the user

### PR DESCRIPTION
**Pull request by xquanluu**
#935

### CODE REVIEW

**Comment by vdharashive** [comment]
@xquanluu now all calls are going to registered user sbc 

We can merge the code

File: lib/utils/network.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+/**
+ *
+ * @param {*} logger
+ * @param {*} hostport
+ * @param {*} protocol
+ * @returns array with bellow properties
```
Comment (davehorton) [inline lib/utils/network.js:7]: please fix typo "below"

File: lib/utils/network.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * 0: protocol
+ * 1: ip address
+ * 2: port
```
Comment (davehorton) [inline lib/utils/network.js:10]: its actually not clear from this description what this function does. It seems to be filtering so as to return the ip address that the feature server is listening on for a given protocol.  Can you make the description state the purpose more clearly?

File: lib/http-routes/api/create-call.js [hunk 3]
```diff
@@ -140,6 +141,16 @@ router.post('/',
         }
       }
 
+      // find handling sbc sip for called user
+      if (target.type === 'user') {
+        const { registrar }  = srf.locals.dbHelpers;
+        const reg = await registrar.query(target.name);
+        if (reg) {
+          sbcAddress = selectHostPort(logger, reg.sbcAddress, 'tcp')[1];
```
Comment (davehorton) [inline lib/http-routes/api/create-call.js:149]: Hmm, I am not loving this.  Feature server already selects an SBC to send to (prior to this change) and I believe it finds the IP address of the SBC in the VPC (the private address).  How is that done?  If different than this, why?

**Review by davehorton — COMMENTED** [review]
I made some comments and additionally I think this feature should be enabled only when a env is set, something like JAMBONES_DIAL_SBC_FOR_REGISTERED_USER

File: lib/utils/network.js (hunk 1 above)
Comment (xquanluu) [reply lib/utils/network.js:7]: fixed

File: lib/utils/network.js (hunk 2 above)
Comment (xquanluu) [reply lib/utils/network.js:10]: fixed

File: lib/http-routes/api/create-call.js (hunk 3 above)
Comment (xquanluu) [reply lib/http-routes/api/create-call.js:149]: @davehorton 

feature server look at redis for keys default:active-sip to get list of sbc private ip address.

then select next sbc address from the list to send request to. this way, feature server cannot send INVITE request to correct sbc which is handling for the user.

this code will get registration from redis for the user, get sbc_address which is handling for the user and as you know that there is a tcp interface from drachtio server will contains private IP address for the sbc. feature server use this private IP to send the request to that SBC.

Comment (davehorton) [reply lib/http-routes/api/create-call.js:150]: its ugly but for now I'll accept.  We need to come back and fix this at some point -- simplify the learning of sbc private/internal address

**Review by davehorton — APPROVED** [review]

**Pull request merged**